tests/common/value_helper.h

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2020-2022 The Khronos Group Inc.
+//  Copyright:	(c) 2017 by Codeplay Software LTD. All Rights Reserved.
+//
+//  This file contains helper functions for modifying and comparing values,
+//  arrays, and objects that implements operator[]
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_COMMON_VALUE_HELPER_H
+#define __SYCLCTS_TESTS_COMMON_VALUE_HELPER_H
+#include "../../util/type_traits.h"
+
+namespace value_helper {
```

Comment:
```suggestion
#include "../../util/type_traits.h"

#include <cassert>

namespace value_helper {
```
BTW, just as an option - we might use `REQUIRE()` from Catch2 for preconditions check at runtime instead of `assert` to avoid dependency on `NDEBUG` macro and to have more clear output on which exact test case triggered a failure

Reply:
Good idea!

Reply:
I have a few concerns about this solution:
- If `value_helper` functions will be used outside of a `TEST_CASE` scope, an exception will be thrown
- Do we really want to depend from catch2 in places where it can be easily avoided?

Reply:
Oops, sorry, looks like I need to clarify. I do not suggest to remove `assert` from entire codebase. I'm just assuming this specific code is designed to be used within `TEST_CASE` scope.

Reply:
I didn't intend to use this code only with catch2, so I think it's not a good idea to add a dependency from the `TEST_CASE` scope.